java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -92,11 +103,36 @@ public static ArrowFlightSqlClientHandler createNewHandler(final FlightClient cl
    * @param flightInfo The {@link FlightInfo} instance from which to fetch results.
    * @return a {@code FlightStream} of results.
    */
-  public List<FlightStream> getStreams(final FlightInfo flightInfo) {
-    return flightInfo.getEndpoints().stream()
-        .map(FlightEndpoint::getTicket)
-        .map(ticket -> sqlClient.getStream(ticket, getOptions()))
-        .collect(Collectors.toList());
+  public List<FlightStream> getStreams(final FlightInfo flightInfo) throws SQLException {
+    final List<FlightStream> allStreams = new ArrayList<>();
+
+    for (final FlightEndpoint endpoint : flightInfo.getEndpoints()) {
+      List<Location> locations = endpoint.getLocations();
+      if (locations.isEmpty()) {
+        allStreams.add(sqlClient.getStream(endpoint.getTicket(), getOptions()));
+        continue;
+      }
+      final Location location = locations.get(0); // purposefully discard other locations
+
+      logger.info(String.format("Getting a client for location %s.", location));
```

Comment:
Don't use String.format() with loggers. This executes the work to generate a string, even if logging is off.
Use the logger directly. Note that placeholders are curly braces in SLF4J eg:
logger.info("Getting client for location {}.", location);

please fix this throughout the PR.